### PR TITLE
Create test snapshots & upgrade to `testthat` 3rd edition

### LINKED ISSUE #784: Update table (and all) tests to check also correct indentation
> Update tests to check also correct indentation (cmt01 for example differs from others as the specific values are right indented by 1 space).
The following tables for example have different indentations than others of the same kind (see original PR for the full list). These need to be properly checked.

- [ ] cmt01 (indentation is shifted to the right in second split -> different from other tables, to check)
- [ ] aet06 (shifted indentation as in cmt01 but only for variant 1. Variant 3 and 5 have normal indentation)

_Originally posted by @Melkiades in https://github.com/insightsengineering/tern/pull/776_

Examples of different (untested) behavior (from cmt01). The first is current while the second is the one that the majority of tables have.
Screenshot 2022-12-20 181039
Screenshot 2022-12-20 181108

@shajoezhu fyi

### PR DESCRIPTION
**Pull request by edelarua**
Closes #784 and #656

### CODE REVIEW

File: tests/testthat/test-compare_variables.R [hunk 1]
```diff
@@ -38,39 +38,39 @@ testthat::test_that("s_compare for factor handles explicit NAs as expected", {
...
-  testthat::expect_equal(result_with_na$pval, 0.9063, tol = 1e-4)
+  testthat::expect_equal(result_with_na$pval, 0.9063, tolerance = 1e-4)
 })
 
 testthat::test_that("s_compare for character works as expected", {
   x <- c("a", "a", "b", "c", "a")
   y <- c("a", "b", "c")
-  result <- testthat::expect_warning(
-    s_compare(x, .ref_group = y, .in_ref_col = FALSE, .var = "x"),
+  suppressWarnings(expect_warning(
```
Comment (Melkiades) [inline tests/testthat/test-compare_variables.R:55]: `testthat::expect_warning`

File: tests/testthat/test-estimate_incidence_rate.R [hunk 2]
```diff
@@ -98,7 +98,7 @@ testthat::test_that("s_incidence_rate works as expected with healthy input", {
     rate_ci = formatters::with_label(c(19.40154, 100.50487), "90% CI")
   )
 
-  testthat::expect_equal(result, expected, tolerance = 1e-4, check.attributes = TRUE)
+  testthat::expect_equal(result, expected, tolerance = 1e-4, ignore_attr = TRUE)
```
Comment (Melkiades) [inline tests/testthat/test-estimate_incidence_rate.R:101]: Here it asked for `check.attributes = TRUE`. Should it be `ignore_attr = FALSE`?

File: tests/testthat/test-g_step.R [hunk 3]
```diff
@@ -90,14 +90,14 @@ testthat::test_that("tidy.step gives expected warnings when there are NAs in y v
     control = control_summarize_vars(conf_level = 0.9),
     variables = list(biomarker = "bla")
   )
-  testthat::expect_warning(
+  suppressWarnings(expect_warning(
```
Comment (Melkiades) [inline tests/testthat/test-g_step.R:93]: `testthat::`

File: tests/testthat/test-g_step.R [hunk 4]
```diff
@@ -90,14 +90,14 @@ testthat::test_that("tidy.step gives expected warnings when there are NAs in y v
...
     broom::tidy(step_matrix),
     "Missing values in the point estimate or CI columns"
-  )
-  testthat::expect_warning(
+  ))
+  suppressWarnings(expect_warning(
```
Comment (Melkiades) [inline tests/testthat/test-g_step.R:97]: `testthat::`

File: tests/testthat/test-g_step.R [hunk 5]
```diff
@@ -107,12 +107,12 @@ testthat::test_that("tidy.step gives expected warnings when there are very large
     control = control_summarize_vars(conf_level = 0.9),
     variables = list(biomarker = "bla")
   )
-  testthat::expect_warning(
+  suppressWarnings(expect_warning(
```
Comment (Melkiades) [inline tests/testthat/test-g_step.R:110]: `testthat::`

File: tests/testthat/test-h_step.R [hunk 6]
```diff
@@ -46,12 +46,12 @@ testthat::test_that("h_step_window works as expected for actual biomarker values
...
   control <- control_step(use_percentile = FALSE, num_points = 5L, bandwidth = 3)
   result <- testthat::expect_silent(h_step_window(x = x, control = control))
-  testthat::expect_is(result, "list")
+  testthat::expect_type(result, "list")
   testthat::expect_named(result, c("sel", "interval"))
   sel <- result$sel
   testthat::expect_true(all(colSums(sel) %in% (control$bandwidth * 2 + c(0L, 1L))))
   interval <- result$interval
-  testthat::expect_is(interval, "matrix")
+  testthat::expect_true(is.matrix(interval))
```
Comment (Melkiades) [inline tests/testthat/test-h_step.R:54]: Does this not work as for list with `expect_type(interval, "matrix")`?

File: tests/testthat/test-summarize_glm_count.R [hunk 7]
```diff
@@ -369,12 +369,11 @@ testthat::test_that("s_glm_count works with no reference group selected.", {
     rate_ratio_ci = formatters::with_label(c(0.6120999, 0.5219551, 1.7749272, 1.5564142), "95% CI"),
     pval = formatters::with_label(c(0.8786994, 0.7093289), "p-value")
   )
-
   testthat::expect_equal(result, expected, tolerance = 0.0000001)
 })
 
 testthat::test_that("s_glm_count fails wrong inputs", {
-  testthat::expect_error(
+  testthat::expect_error(s_glm_count(
```
Comment (Melkiades) [inline tests/testthat/test-summarize_glm_count.R:376]: Wait, so before there was no function here? ahah

File: tests/testthat/test-table_adat03.R [hunk 8]
```diff
@@ -67,23 +67,6 @@ testthat::test_that("ADAT03 is produced correctly", {
...
-      "", "", "", "CV (%)", "", "", "89.4", "", "10.0", "", "", "84.3", "", "35.3", "77.7",
-      "", "", "", "Geometric Mean", "", "", "NA", "", "1.338e+01", "", "", "NA", "", "1.891e+01", "NA",
-      "", "Samples with", "Concentration", "≤ 15μg/mL {2}", "", "", "738 (78.7%)", "", "228 (85.1%)", "", "",
-      "836 (52.8%)", "", "210 (39.8%)", "2012 (60.6%)"
-    ),
-    .Dim = c(15L, 10L)
-  )
-  testthat::expect_identical(result_matrix, expected_matrix)
+  res <- testthat::expect_silent(result)
+  testthat::expect_snapshot(res)
```
Comment (Melkiades) [inline tests/testthat/test-table_adat03.R:71]: So clean! Well done

File: tests/testthat/test-g_forest.R [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+      col_x = 1,
+      col_ci = 2,
+      xlim = c(0.5, 2),
+      x_at = c(0.5, 1, 2),
+      vline = 1,
+      forest_header = c("Hello", "World"),
+      draw = FALSE
+    )
+  )
+})
```
Comment (Melkiades) [inline tests/testthat/test-g_forest.R:53]: Where was this test? I guess it was in a table test

Comment (Melkiades) [reply tests/testthat/test-g_forest.R:53]: Ah ok, it is new. Was it not tested??

**Review by Melkiades — CHANGES_REQUESTED** [review]
Really amazing job Emily! With this, I think we can split scda in a couple of days during next week. Also, well done including `testthat` 3! I think you need to specify it also in the DESCRIPTION (Set the `Config/testthat/edition` field to 3).

Fix the error in the new test:
```
  ── Error ('test-g_forest.R:4'): (code run outside of `test_that()`) ────────────
  <dplyr:::mutate_error/rlang_error/error/condition>
  Error in `dplyr::mutate(., ARM = fct_relevel(ARM, "B: Placebo"), rsp = AVALC == 
      "CR")`: Problem while computing `ARM = fct_relevel(ARM, "B: Placebo")`.
  Caused by error in `fct_relevel()`:
  ! could not find function "fct_relevel"
```
The other comments are not super necessary but take a look at them, they are fast fixes.

I read (beside the guide from testthat itself), this [summary guide](https://www.tidyverse.org/blog/2022/02/upkeep-testthat-3/), and I think we should add a note to the future `tern` tests that highlights the need of proper test naming. I think it is already almost there. We need just to make sure of this (I think with `test-g_forest.R` you moved already  towards this).

File: tests/testthat/test-h_step.R (hunk 6 above)
Comment (Melkiades) [reply tests/testthat/test-h_step.R:54]: Screenshot 2023-01-19 105049
Ok, never mind; it seems it is the only way :D

File: tests/testthat/test-g_forest.R (hunk 9 above)
Comment (edelarua) [reply tests/testthat/test-g_forest.R:53]: @Melkiades yeah, I'm not sure why there was a big drop in code coverage for `g_forest` when nothing was really changed but I added these tests. I'll try to improve it more.

File: tests/testthat/test-summarize_glm_count.R (hunk 7 above)
Comment (edelarua) [reply tests/testthat/test-summarize_glm_count.R:376]: Yeah this file was very weird!

**Review by Melkiades — APPROVED** [review]
Lgtm!! Great, great job @edelarua ^^

fyi @shajoezhu

**Comment by shajoezhu** [comment]
AMAZING! Thanks so much @edelarua

**Pull request merged**